File: 채현/W1/큰 수 만들기.py
def solution(number, k):
    stack = []

    for i in number:
        while k > 0 and stack and stack[-1] < i:
            stack.pop()
            k -= 1
        stack.append(i)

    # 아직 k가 남아 있으면 뒤에서 제거
    if k > 0:
        stack = stack[:-k]

    return ''.join(stack)

File: 주은/W1/큰_수_만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    int len = number.length();
    for (int i = 0; i < len; i++) {
        while(answer.length() != 0 && k > 0 && answer.back() < number[i]) {
            answer.pop_back();
            k--;
        }
        answer += number[i];
    }
    
    while (k > 0) {
        answer.pop_back();
        k--;
    }
    return answer;
}